src/provider/provider_cuda.c

```diff
@@ -546,15 +546,27 @@ static void cu_memory_provider_get_last_native_error(void *provider,
                               &error_string);
 
     size_t buf_size = 0;
-    strncpy(TLS_last_native_error.msg_buff, error_name, TLS_MSG_BUF_LEN - 1);
+    if (error_name) {
+        strncpy(TLS_last_native_error.msg_buff, error_name,
+                TLS_MSG_BUF_LEN - 1);
+    } else {
+        strncpy(TLS_last_native_error.msg_buff, "cuGetErrorName() failed",
```

Comment:
nit: can we extend this error message, or/and add comment, that it can happen if cuda is allready destroyed. 

documentation for this functions, says that they never return NULL, so this might be misleading

Reply:
Done